images/Images/ImageRegrid.tcc

```diff
@@ -1263,63 +1263,133 @@ void ImageRegrid<T>::make2DCoordinateGrid(LogIO& os, Bool& allFailed, Bool&misse
+          // to find the interpolated result at.  (,,0) pertains to
+          // inX and (,,1) to inY
+          in2DPos(i,j,0) = inPixel(inXIdx);
+          in2DPos(i,j,1) = inPixel(inYIdx);
+          allFailed = False;
+          succeed(i,j) = True;
+          //
+          if (decimate <= 1) {
+            minInX = min(minInX,inPixel(inXIdx));
+            minInY = min(minInY,inPixel(inYIdx));
+            maxInX = max(maxInX,inPixel(inXIdx));
+            maxInY = max(maxInY,inPixel(inYIdx));
+          } else {
+            iInPos2D(ii,jj) = inPixel(inXIdx);
+            jInPos2D(ii,jj) = inPixel(inYIdx);
+            ijInMask2D(ii,jj) = True;
+          };
+        };
+      };
+    };
+  } else {
+    // generate coordinate conversions in bulk
+    // set storage matrices for the conversions
+    Matrix<Double> inPixelMatrix(nPixelAxes,nConversions);
+    Matrix<Double> outPixelMatrix(nPixelAxes,nConversions);
+    Matrix<Double> worldMatrix(nPixelAxes,nConversions);
+    Vector<Bool> failures1(nConversions);
+    Vector<Bool> failures2(nConversions);
+    // set the output coordinates
+    uInt kk = 0;
+    jj = 0;
+    for (uInt j=0; j<nj; j+=jInc,jj++) {
+      ii = 0;
+      for (uInt i=0; i<ni; i+=iInc,ii++) {
+        outPixelMatrix(outXIdx,kk) = i + outPos[xOutAxis];
+        outPixelMatrix(outYIdx,kk) = j + outPos[yOutAxis];
+        kk++;
+      };
+    };
+    // do the conversions
+    if (isDir) {
+      ok1 = outDir.toWorldMany( worldMatrix, outPixelMatrix, failures1 );
+      ok2 = False;
+      if (ok1) ok2 = inDir.toPixelMany( inPixelMatrix, worldMatrix, failures2 );
+    } else {
+      ok1 = outLin.toWorldMany( worldMatrix, outPixelMatrix, failures1 );
+      ok2 = False;
+      if (ok1) ok2 = inLin.toPixelMany( inPixelMatrix, worldMatrix, failures2 );
+    }
+    // only keep going if some of the conversions succeeded
+    if (!ok2) {
+      allFailed = True;
+      succeed.set(False);
+      ijInMask2D.set(False);
+    } else {
+      allFailed = False;
+      kk = 0;
+      jj = 0;
+      for (uInt j=0; j<nj; j+=jInc,jj++) {
+        ii = 0;
+        for (uInt i=0; i<ni; i+=iInc,ii++) {
+          if (failures1(kk) || failures2(kk)) {
+            succeed(i,j) = False;
+            if (decimate>1) ijInMask2D(ii,jj) = False;
+          } else {
+            // This gives the 2D input pixel coordinate (relative to
+            // the start of the full Lattice)
+            // to find the interpolated result at.  (,,0) pertains to
+            // inX and (,,1) to inY
+            in2DPos(i,j,0) = inPixelMatrix(inXIdx,kk);
+            in2DPos(i,j,1) = inPixelMatrix(inYIdx,kk);
+            succeed(i,j) = True;
+            //
+            if (decimate <= 1) {
+              minInX = min(minInX,inPixelMatrix(inXIdx,kk));
+              minInY = min(minInY,inPixelMatrix(inYIdx,kk));
+              maxInX = max(maxInX,inPixelMatrix(inXIdx,kk));
+              maxInY = max(maxInY,inPixelMatrix(inYIdx,kk));
```

Comment:
This looks like a copy-pase error, and should probably compute `minInY` instead.

Reply:
Could be cabin-fever but these seem ok to me.

Reply:
Yes, the cabin-fever was on my end.